src/tojson.jl

```diff
@@ -72,69 +79,109 @@ function tojson(output::String, ts::ReportingTestSet)
         end
     end
 
-    # Flag set in walk!(), used for top-level status property
+    # Flag set in push_result!(), used for top-level status property
     any_failed = false
+    # All stdout from the top level test set, used for all tests.
     output = truncate_output(output)
 
     """
-        walk!(prefix, tests, testset)
+        push_result!(tests, result, name)
+
+    Push a test result called `name` to `tests`. Returns the pushed Dict or `nothing` if the test was not pushed.
+    """
+    function push_result!(tests, result::Test.Result, name)
+        status = nothing
+        message = nothing
+        test_code = nothing
+
+        if result isa Test.Pass
+            status = "pass"
+            test_code = "@test $(result.orig_expr)"
+        elseif result isa Test.Fail
+            status = "fail"
+            message = string(result)
+            test_code = "@test $(result.orig_expr)"
+        elseif result isa Test.LogTestFailure
+            status = "fail"
+            message = string(result)
+            test_code = "@test_logs $(join(result.patterns, ' ')) $(result.orig_expr)"
+        elseif result isa Test.Error
+            status = "error"
+            message = result.backtrace
+            test_code = "@test " * result.orig_expr
```

Comment:
```suggestion
            test_code = "@test $(result.orig_expr)"
```

For consistency with the branches above.